### DISCUSSION TITLE
Is there a way to use React components natively in mint?

### DISCUSSION THREAD

**Discussion by mrjones2014** [mint-lang/mint#350] [Q&A] [answered]
I have a robust Rich Text Editor component I built over a long time in React, I'm wondering if it's possible to use my React rich text editor in Mint natively.

Mint looks awesome and I'm really excited to try it out!

**Comment by gdotdesign** [comment, marked as answer]
It's not trivial but it can be done, you will need to have a custom build for it:

1. You need to change the [runtime](https://github.com/mint-lang/mint-runtime) to include your component under the Mint global object (https://github.com/mint-lang/mint-runtime/blob/master/source/Main.js#L46 `MyComponent: MyComponent`)
2. Build the `runtime.js` file and copy it into the code of the compiler https://github.com/mint-lang/mint/blob/master/src/assets/runtime.js
3. Expose a reference to the component in the [`wrap_runtime`](https://github.com/mint-lang/mint/blob/master/src/compilers/top_level.cr#L219) method (`const MyComponent = mint.MyComponent`)
4. Build the `mint` binary
5. You can use your component using the reference and inline JavaScript `_h(MyComponent, props, children)`

A way to interoperate with NPM will be possible before 1.0.

**Comment by mrjones2014** [comment]
@gdotdesign thanks! I'll close this issue then.
